src/MongoDB.Driver.Core/Core/Connections/TcpStreamFactory.cs

```diff
@@ -181,9 +181,7 @@ private async Task ConnectAsync(Socket socket, EndPoint endPoint, CancellationTo
                 try
                 {
                     var dnsEndPoint = endPoint as DnsEndPoint;
-#if NETSTANDARD1_5 || NETSTANDARD2_0
-                    await socket.ConnectAsync(endPoint).ConfigureAwait(false);
-#else
+#if NET452
```

Comment:
Don't reverse the then/else parts to make the diff smaller.

Reply:
done